index.js

```diff
@@ -142,54 +156,169 @@ function updateSchemesAndHost(req) {
+ *
+ * @description `response` middleware.
+ *
+ * @param {Express} expressApp - the express app
+ *
+ * @param {Object} [options] optional configuration options
+ * @param {string|undefined} [options.pathToOutputFile=undefined] where to write the openAPI specification to.
+ * Specify this to create the openAPI specification file.
+ * @param {number} [options.writeIntervalMs=10000] how often to write the openAPI specification to file
+ *
+ * @returns void
+ */
+function injectResponseMiddleware(expressApp, options = { pathToOutputFile: undefined, writeIntervalMs: 1000 * 10 }) {
+  /**
+   * save the `expressApp` to our local `app` variable.
+   * Used here, but not in `injestRequestMiddleware`,
+   * because this comes before it.
+   */
+  app = expressApp;
+
+  const { pathToOutputFile, writeIntervalMs } = options;
 
-  // middleware to handle responses
+  /** middleware to handle RESPONSES */
+  // eslint-disable-next-line complexity
   app.use((req, res, next) => {
     try {
       const methodAndPathKey = getMethod(req);
+
       if (methodAndPathKey && methodAndPathKey.method) {
         processors.processResponse(res, methodAndPathKey.method);
       }
-      let firstTime = true;
+
+      let firstTime = true; /** run instantly the first time. TODO do not set set `lastRecordTime` at the start until we run */
       const ts = new Date().getTime();
-      if (firstTime || aPath && ts - lastRecordTime > writeInterval) {
+
+      if (firstTime || pathToOutputFile && ts - lastRecordTime > writeIntervalMs) {
         firstTime = false;
         lastRecordTime = ts;
-        fs.writeFile(aPath, JSON.stringify(spec, null, 2), 'utf8', err => {
-          const fullPath = path.resolve(aPath);
+
+        fs.writeFile(pathToOutputFile, JSON.stringify(spec, null, 2), 'utf8', err => {
+          const fullPath = path.resolve(pathToOutputFile);
+
           if (err) {
             throw new Error(`Cannot store the specification into ${fullPath} because of ${err.message}`);
           }
         });
       }
-    } catch (e) {}
-    next();
+    } catch (e) {
+      /** TODO - shouldn't we do something here? */
+    } finally {
+      /** always call the next middleware */
+      next();
+    }
   });
+}
 
-  // make sure we list routes after they are configured
-  setTimeout(() => {
-    // middleware to handle requests
-    app.use((req, res, next) => {
-      try {
-        const methodAndPathKey = getMethod(req);
-        if (methodAndPathKey && methodAndPathKey.method && methodAndPathKey.pathKey) {
-          const method = methodAndPathKey.method;
-          updateSchemesAndHost(req);
-          processors.processPath(req, method, methodAndPathKey.pathKey);
-          processors.processHeaders(req, method, spec);
-          processors.processBody(req, method);
-          processors.processQuery(req, method);
-        }
-      } catch (e) {}
+module.exports.injectResponseMiddleware = injectResponseMiddleware;
```

Comment:
could you please move all exports to the end of the file and maybe refactor it a little like that:

```javascript
module.exports = {
  injectResponseMiddleware: ...
  injectRequestMiddleware: ...
  ...
};
```

Reply:
Sure thing!